lib/query.js

```diff
@@ -392,9 +392,27 @@ function Query (client, ns, set, options) {
   /**
    * If set to a valid serialized query, calling {@link query#foreach} will allow the next page of records to be queried while preserving the progress
    * of the previous query. If set to <code>null</code>, calling {@link query#foreach} will begin a new query.
-   * @member {number} Query#queryState
+   * @member {Object} Query#queryState
    */
   this.queryState = undefined
+
+  /**
+   * The time-to-live (expiration) of the record in seconds.
+   * There are also special values that can be set in the record TTL:
+   *
+   * 0 (defined Aerospike.ttl.NAMESPACE_DEFAULT), which means that the
+   * record will adopt the default TTL value from the namespace.
+   *
+   * -1 (defined Aerospike.ttl.NEVER_EXIRE), which means that the record
+   * will get an internal "void_time" of zero, and thus will never expire.
+   *
+   * -2 (defined Aerospike.ttl.DONT_UPDATE), which means that the record
+   * ttl will not change when the record is updated.
+   *
+   * Note that the TTL value will be employed ONLY on background query writes.
+   * @member {number} Query#ttl
+   */
+  this.ttl = 0
```

Comment:
Is there a missing field `.ttl` in options? Same for Scan

Reply:
Just added changes to allow the TTL to be set in query and scan options as well.  Added tests to verify correct behavior on scans and queries. This was a nice find.  Thank you!